### PR TITLE
Feature/backport carma wm to ros2

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
This PR backports to ROS2 the fixes for carma_wm (ROS1) which were identified during 4.0 release testing. In addition, it reverts one change in this PR https://github.com/usdot-fhwa-stol/carma-platform/pull/1678  which invalidated a unit test by setting expected lanelet ids to values which were not actually set anywhere. (This is the change set shown in SignalizedIntersectionManagerTest.cpp) Also fixes a unit test in the same file where the ingress and egress ids were swapped as that was fixed in verification and not applied to the test. 

## Related Issue
Resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1740

## Motivation and Context
Fixes applied both in ROS1 and ROS2 when appropriate

## How Has This Been Tested?
Build and unit testing. Achieves unit test parity with 4.0

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: carma_wm_ros2/include/carma_wm_ros2/collision_detection.hpp [hunk 1]
```diff
@@ -42,10 +42,12 @@ namespace carma_wm {
         typedef boost::geometry::model::point<double, 2, boost::geometry::cs::cartesian> point_t;
         typedef boost::geometry::model::polygon<point_t> polygon_t;
 
+        //TODO: verify correctness of descriptions for each of these elements
```
Comment (kjrush) [inline carma_wm_ros2/include/carma_wm_ros2/collision_detection.hpp:45]: Is this todo still open?

**Review by kjrush — APPROVED** [review]

File: carma_wm_ros2/include/carma_wm_ros2/collision_detection.hpp (hunk 1 above)
Comment (msmcconnell) [reply carma_wm_ros2/include/carma_wm_ros2/collision_detection.hpp:45]: Yeah both of these TODOs are still open left from John's compile warnings PR

File: carma_wm_ros2/src/collision_detection.cpp [hunk 2]
```diff
@@ -22,7 +22,10 @@ namespace carma_wm {
 
     namespace collision_detection {
 
-        std::vector<carma_perception_msgs::msg::RoadwayObstacle> WorldCollisionDetection(const carma_perception_msgs::msg::RoadwayObstacleList& rwol, const carma_planning_msgs::msg::TrajectoryPlan& tp, const geometry_msgs::msg::Vector3& size, const geometry_msgs::msg::Twist& veloctiy,const __uint64_t target_time) {
+        //TODO: consider rewriting this method to improve efficiency; it has object_count*prediction_count*trajectory_point sqrt calls!
```
[the remark this answers is not in the document]
Comment (msmcconnell) [reply carma_wm_ros2/src/collision_detection.cpp:25]: Agree

File: carma_wm_ros2/test/CARMAWorldModelTest.cpp [hunk 3]
```diff
@@ -1377,7 +1377,7 @@ TEST(CARMAWorldModelTest, processSpatFromMsg)
   cmw.processSpatFromMsg(spat);
   lights1 = cmw.getMutableMap()->laneletLayer.get(ll_1.id()).regulatoryElementsAs<lanelet::CarmaTrafficSignal>();
   // partial state 3
-  EXPECT_EQ(lanelet::time::durationFromSec(45), lights1[0]->fixed_cycle_duration);
+  EXPECT_EQ(lanelet::time::durationFromSec(44.5), lights1[0]->fixed_cycle_duration); // NOTE: 44.5 hear instead of 45 because of the intentional subtraction of 0.5s inside the implementation
```
Comment (kjrush) [inline carma_wm_ros2/test/CARMAWorldModelTest.cpp:1380]: Typo, hear->here.

File: carma_wm_ros2/test/CARMAWorldModelTest.cpp [hunk 4]
```diff
@@ -1389,7 +1389,7 @@ TEST(CARMAWorldModelTest, processSpatFromMsg)
   cmw.processSpatFromMsg(spat);
   lights1 = cmw.getMutableMap()->laneletLayer.get(ll_1.id()).regulatoryElementsAs<lanelet::CarmaTrafficSignal>();
   // and query the regem again to check if its entries are updated, by checking revision or getState or predictState etc
-  EXPECT_EQ(lanelet::time::durationFromSec(45), lights1[0]->fixed_cycle_duration);
+  EXPECT_EQ(lanelet::time::durationFromSec(44.5), lights1[0]->fixed_cycle_duration); // NOTE: 44.5 hear instead of 45 because of the intentional subtraction of 0.5s inside the implementation
```
Comment (kjrush) [inline carma_wm_ros2/test/CARMAWorldModelTest.cpp:1392]: Same as above.

**Review by kjrush — APPROVED** [review]

**Review by kjrush — APPROVED** [review]

**Pull request merged**